src/probnum/filtsmooth/gaussfiltsmooth/extendedkalman.py

```diff
@@ -17,7 +17,9 @@ def __init__(self, non_linear_sde, num_steps):
         self.num_steps = num_steps
         super().__init__()
 
-    def transition_realization(self, real, start, stop, linearise_at=None, **kwargs):
+    def transition_realization(
+        self, real, start, stop, linearise_at=None, diffusion=1.0, **kwargs
```

Comment:
Do you have any preferences for the ordering of parameters? 

Edit: Looking into at it now, I would maybe move the diffusion parameter before `linearise_at` since it appears in all transition functions.

Reply:
I think I agree. 

Two minor questions here: 1. do we want a default set to 1 or a None (for documentation/communication reasons I think this is a valid question); related: 2. do we want to name it "diffusion" or something like "special_diffusion" to communicate that this should only be touched if one knows what is going on?

Reply:
Intuitively I would name it `diffusion` or maybe `_diffusion` and set it to `1.0`, since that makes it explicit what's actually happening. I mean it _is_ the diffusion, and we set it to value 1?

Though I understand your motivation for having a different name to make clear that that's a very custom functionality that most people should not care about, and in that line of thought I also see the appeal of setting it to `None`.

So, what do you think?

Reply:
To brainstorm names a bit: `custom_diffusion`? `internal_diffusion`? `diffusion_coefficient`/`diffcoeff`?

Reply:
let's go for `_diffusion` (and likewise, `_intermediate_step` and perhaps also `_linearise_at`)? A default to 1 is fine. It keeps the code simple.

Reply:
Ok, I renamed all three of them to have leading underscores.